internal/cluster/proxy.go

```diff
@@ -0,0 +1,156 @@
+package cluster
+
+import (
+	"crypto/md5" // #nosec
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/asaskevich/govalidator"
+	"github.com/go-openapi/swag"
+	"github.com/pkg/errors"
+)
+
+// Proxy defines proxy configuration
+type Proxy struct {
+	HTTPProxy  *string
+	HTTPSProxy *string
+	NoProxy    *string
+}
+
+// NewProxy constructs a proxy instance
+func NewProxy(httpProxy, httpsProxy, noProxy *string) *Proxy {
+	return &Proxy{
+		HTTPProxy:  httpProxy,
+		HTTPSProxy: httpsProxy,
+		NoProxy:    noProxy,
+	}
+}
+
+// IsSet returns true if either HTTP, or HTTPS, or
+// both were provided
+func (p *Proxy) IsSet() bool {
+	return (p.HTTPProxy != nil && *p.HTTPProxy != "") ||
+		(p.HTTPSProxy != nil && *p.HTTPSProxy != "")
+}
+
+// Validate validates the proxy configuration
+func (p *Proxy) Validate() error {
+
+	http := swag.StringValue(p.HTTPProxy)
+	if http != "" {
+		if err := validateHTTPProxyFormat(http); err != nil {
+			return errors.Errorf("Failed to validate HTTP Proxy: %s", err)
+		}
+	}
+
+	https := swag.StringValue(p.HTTPSProxy)
+	if https != "" {
+		if err := validateHTTPProxyFormat(https); err != nil {
+			return errors.Errorf("Failed to validate HTTPS Proxy: %s", err)
+		}
+	}
+
+	noProxy := swag.StringValue(p.NoProxy)
+	if noProxy != "" {
+		if err := validateNoProxyFormat(noProxy); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Diff detects if two proxy configurations differ.
+// Changed proxy settings mean that a new ISO file must be generated
+// to include the updated proxy settings
+func (p *Proxy) Diff(httpProxy, httpsProxy, noProxy string) bool {
+	if httpProxy != swag.StringValue(p.HTTPProxy) ||
+		httpsProxy != swag.StringValue(p.HTTPSProxy) ||
+		noProxy != swag.StringValue(p.NoProxy) {
+		return true
```

Comment:
This will return true even if no values where send via the REST API (i.e. no changes)